xija/model.py

```diff
@@ -103,9 +103,10 @@ class XijaModel(object):
     def __init__(self, name=None, start=None, stop=None, dt=None,
                  model_spec=None, cmd_states=None, evolve_method=None,
                  rk4=None, limits=None):
-        # If model_spec supplied as a string then read model spec as a dict
-        if isinstance(model_spec, six.string_types):
+        # If model_spec is a str or Path then read that file
+        if isinstance(model_spec, (str, Path)):
```

Comment:
This makes sense to me.  I can't think of a use case where we'd have something that was recognized as a in six.string_types that wouldn't be fine as a str.  I think my previous question still stands though, are there other classes that would be just as reasonable for a "thing to read for the model spec" as a Path or a str?  And if so, should this just try the read and work by duck typing or does the explicit handling here cover all reasonable cases?  Thanks!

Reply:
There is always that question. I actually spent some time looking at `open` and the `os.PathLike` protocol. So you are right that a more perfect solution is a try/except after determining that the input model_spec is not None and not a dict (but what if it is a dict-like object that doesn't inherit from dict??).  Except that it ends up being more complex code.

In the end I think this is good enough, and if not we'll fix it then. The user base for this class is quite small.

Reply:
Thanks for the conversation about it.